packages/material-ui-styles/src/makeStyles/makeStyles.d.ts

```diff
@@ -1,21 +1,21 @@
 import {
   ClassNameMap,
-  PropsOfStyles,
   Styles,
   WithStylesOptions,
+  StyledComponentProps,
 } from '@material-ui/styles/withStyles';
 import { Omit } from '@material-ui/types';
 import { DefaultTheme } from '../defaultTheme';
 
+export type MakeStylesHook<
+  Props extends {} = {},
+  ClassKey extends string = string
+> = keyof Props extends never
+  ? (props?: Pick<StyledComponentProps<ClassKey>, 'classes'>) => ClassNameMap<ClassKey>
```

Comment:
We shouldn't re-use this type here. A simple `{ classes: ClassKeyMap<ClassKey> }` should be enough.

Reply:
Why we should not? All props will be in concistence state. If some body decides to change `classes` to `classList` (after source code changed for instance) TS will notify about it. It's more safe way I think.

Reply:
Hah, we have already imported ClassKeyMap.
Refactored.